### ISSUE TITLE
gui : no vumeter on vst2 in OBS STUDIO 27.2.4 64 bits

### ISSUE THREAD

**Issue by rafaelholeva** [trummerschlunk/master_me#92]
Hello, I use OBS Studio in its version 27.2.4 (x64) with the latest version of the plugin vst2 master-me (x64) 1.1.0 and also the GUI (vumeter) is not displayed correctly. Also, there is no animation of the levels in VU meter while the audio is played in OBS STUDIO. I hope this resolves itself, as other plugins are working correctly.
Congratulations on the project, I hope that in the future there will be a standalone version to run on windows compatible with ASIO DRIVER and WASAPI and Wave, DirectSound

**Comment by falkTX** [comment]
I am 90% sure this is an issue on OBS plugin handling side.
DPF VST2 based plugins require the host to call `effEditIdle` at regular intervals, something that likely is not happening here.

**Comment by rafaelholeva** [comment]
Well then! The OBS people blame the Plugin! Other plugins already had the same behavior and it was fixed by the developers. Well, could you adjust the plugin to work on OBS? I'm using obs 27 (x64) for windows.

**Comment by falkTX** [comment]
no, because skipping `effEditIdle` is not possible on Linux, and I would prefer to keep the implementations the same if possible.
besides, DPF based VST2 plugins have been a thing for a while, this is the first time to see a host behaving like this.

in any case, I have as a task to do proper audio plugin support in OBS later in the year (not just VST2 but VST3, CLAP, LV2 etc). so it will be a matter of time until this works fully.

**Comment by rafaelholeva** [comment]
We would all be very grateful if you could create a version that is functional in OBS Studio. I believe that most who are going to use it for live, will use it together with OBS, so this is important! As well as an auto-executable version natively on windows, as the current binary requires binary JACK mode and fails to start on Windows x64

**Comment by trummerschlunk** [comment]
@rafaelholeva  I agree, that this is the prominent use case for master_me.

@falkTX can I help with this somehow? Opening an issue over at OBS?

**Comment by falkTX** [comment]
It is not an issue with OBS directly, but with OBS-VST plugin (from what I could tell).

I am finishing up most of the CLAP stuff this month, so next efforts can be for the OBS+carla-based-hosting project.
Worst case I try to allocate some time during christmas/newyear holidays for it.

**Comment by NuclearMonster** [comment]
wanted to note I believe this is still an issue with the latest release (1.2.0) of master_me on macos (14.4.1) and OBS Studio 30.1.0. I've only tested with OBS so I am not sure what the GUI would look like if it were functioning properly, but there is no animation in the VST while the source is being used.

**Comment by falkTX** [comment]
that is still an issue on OBS side.

also, my proposal for plugin support in OBS using carla hosting code was rejected in the end. I will come back to it at some point later to create an unofficial plugin based on the work done already, but not sure when that happens exactly.
